openssl-dynamic/src/main/c/ssl.c

```diff
@@ -1607,6 +1607,54 @@ TCN_IMPLEMENT_CALL(jboolean, SSL, setCipherSuites)(TCN_STDARGS, jlong ssl,
     return rv;
 }
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+/*
+  Backport of SSL_SESSION_get_master_key from 1.1
+*/
+size_t SSL_SESSION_get_master_key(const SSL_SESSION *session,
+                                  unsigned char *out, size_t outlen)
+{
+    if (outlen == 0)
+        return session->master_key_length;
+    if (outlen > session->master_key_length)
+        outlen = session->master_key_length;
+    memcpy(out, session->master_key, outlen);
+    return outlen;
+}
+#endif
+
+
+TCN_IMPLEMENT_CALL(jbyteArray, SSL, getMasterKey)(TCN_STDARGS, jlong ssl)
+{
+
+    SSL *ssl_ = J2P(ssl, SSL *);
+    TCN_CHECK_NULL(ssl_, ssl, NULL);
+
+    SSL_SESSION *session = NULL;
+    if (session == NULL) {
+        return NULL;
+    }
```

Comment:
This does not look correct.. session will also be `NULL` here. Did you forget to call `SSL_get0_SESSION(...)` ?